### PR TITLE
Restart functionality is broken in cmd line version

### LINKED ISSUE #56: Restart functionality is broken in cmd line version
Extend the base Controller class with necessary method signatures for wanted functionality across all controllers. 
Implement a restart method for all controllers.

### PR DESCRIPTION
**Pull request by IKostric**
Fixes #56

### CODE REVIEW

File: moviebot/controller/controller.py [hunk 1]
```diff
@@ -1,25 +1,45 @@
...
-    """This is the main class that controls the other components of the
-    IAI MovieBot. The controller executes the conversational agent."""
+    def __init__(self, configuration: Dict[str, Any]):
+        """This is the main class that controls the other components of the
+        IAI MovieBot. The controller executes the conversational agent.
 
-    def __init__(self):
-        """Initializes some basic structs for the Controller."""
+        Args:
+            configuration: the settings for the agent
```
Comment (WerLaj) [inline moviebot/controller/controller.py:19]: Capital letter and dot at the end.

**Review by WerLaj — APPROVED** [review]
LGTM

File: moviebot/controller/controller.py [hunk 2]
```diff
@@ -1,25 +1,45 @@
...
+        """
+        raise NotImplementedError
 
-        Args:
-            configuration: the settings for the agent
+    def initialize_agent(self) -> Agent:
+        """Initializes and returns an agent based on configuration."""
+        agent = Agent(self.configuration)
+        agent.initialize()
+        print(
```
Comment (NoB0) [inline moviebot/controller/controller.py:34]: Maybe here you can use logging instead of print.

**Review by NoB0 — APPROVED** [review]
LGTM. Just a question regarding the use of logging vs. print.

File: moviebot/controller/controller.py (hunk 1 above)
Comment (IKostric) [reply moviebot/controller/controller.py:19]: Done

**Pull request merged**